### PR TITLE
Views

### PR DESCRIPTION
**Pull request by truetechcode**
Some of the issues you noted with our pull request for the controller branch, we already fixed in the views branch, so we want you to take a look at it before we proceed thanks.

### CODE REVIEW

**Comment by Oghenebrume50** [comment]
we have done everything asides the testing, we would be starting the testing now on a new branch, we want feedback for this, to make sure we are on the right track

File: Gemfile [hunk 1]
```diff
@@ -1,62 +1,30 @@
...
-# Use ActiveStorage variant
-# gem 'mini_magick', '~> 4.8'
-
-# Use Capistrano for deployment
-# gem 'capistrano-rails', group: :development
-
-# Reduces boot times through caching; required in config/boot.rb
+gem 'rails',        '5.2.3'
+gem 'puma',         '3.9.1'
+gem 'bcrypt',       '3.1.12'
```
Comment (rvvergara) [inline Gemfile]: I'm not sure why we still need `bcrypt` here

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class UsersController < ApplicationController
+
+  def index
+    @users = User.all
+  end
+
+  def show
```
Comment (rvvergara) [inline app/controllers/users_controller.rb]: I'm confused about your `UsersController` `show` action. So the only `user` page we will always see is the `current_user` ?

Kindly fix this. The `current_user` should be able to visit all other users' pages.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class PostsController < ApplicationController
+
+  def index
+    @posts = current_user.timeline_posts(current_user).flatten
```
Comment (rvvergara) [inline app/controllers/posts_controller.rb]: This makes us only see `current_user` 's posts. But what if i wanna see other users' posts also?

**Comment by Oghenebrume50** [comment]
> ### The following items haven't been fixed:
> * [ ]  User can comment on posts => **Currently a user can only see his/her own posts. Let's keep working on making the views user friendly**
> * [ ]  User can like posts => **Currently a user can only like on his own posts**
> 
> ### UsersController show action and PostsController `index` action need some serious correction
> Please see my annotations in those files and let's make sure to correct them.
> 
> As a logged in user all i can always visit is my own profile page and can only see my own posts, comment on my own posts and like my own posts. This makes a social-network **unsocial**.
> 
> ### We should also have a way to "unlike" a post
> That means a `destroy` action in the `LikesController`
> 
> Before we go further, let's first make sure the above are fixed.

Hi @rvvergara  I would be making some comments with regards to what I believe is an oversight from your review.

> **Odin project 11:** Treat the Posts Index page like the real Facebook’s “Timeline” feature – show all the recent posts from the current user and users she is friends with.
Currently, the post index page follows the Odin project guidelines and users can only see their post and the post of their friends. and this post can be commented on too, I would attach a screenshot

The above screenshot shows posts that are not only from the current logged in user, and you can see comments and likes, so a user or a friend of a user can see their post, comment on it and like it.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Oghenebrume50) [reply app/controllers/users_controller.rb]: > **Odin project 13:** The User Show page contains their Profile information, photo, and Posts.

Our user show controller is programmed according to this, that was the requirement

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Oghenebrume50) [reply app/controllers/posts_controller.rb]: Like I commented below with the attached screenshot this controller gets the current_user's post and the post of his friends. According to **Odin project, bullet point 13** that is what we are supposed to do, see the attached screenshot below to verify this

**Comment by Oghenebrume50** [comment]
@rvvergara Please, if this is not clear enough I can do a video to show these functionalities, I and my paternal are really confused on how you missed these functionalities we already implemented, plus is there some other requirements, because in the Microverse milestone, there were no extra requirements added, so we just followed the Odin project.

**Comment by rvvergara** [comment]
Using the app in 2 browsers and logged on as 2 different users.

In Chrome:

image

In Firefox:

image

As you will see in both browsers I am in the `localhost:3000/users/2` url. The weird thing is that I am seeing different users for the exact same user_id in that route.

Now here's the suspect code:
image


You will notice what's wrong at first glance.

### Seeing other users' posts in the Timeline

I stand corrected -> Thanks for clarifying

### Adding requirements concern

Please note that the checklist I included in the reviews are official Microverse checklists. My comments revolve around the guidelines and instructions. The issue that is depicted in the images above aren't new features but are requirements that are implicit with the project.

**Pull request merged**